diff --git a/include/v8.h b/include/v8.h
index 573e80176d..0d3e98d079 100644
--- a/include/v8.h
+++ b/include/v8.h
@@ -4318,6 +4334,13 @@ class V8_EXPORT ArrayBuffer : public Object {
      */
     virtual void* AllocateUninitialized(size_t length) = 0;

+    /**
+     * Free the memory block of size |length|, pointed to by |data|.
+     * That memory must be previously allocated by |Allocate| and not yet freed
+     * with a call to |Free| or |Realloc|
+     */
+    virtual void* Realloc(void* data, size_t length) = 0;
+
     /**
      * Free the memory block of size |length|, pointed to by |data|.
      * That memory is guaranteed to be previously allocated by |Allocate|.
diff --git a/src/api.cc b/src/api.cc
index 8b177d041d..180ce3c9de 100644
--- a/src/api.cc
+++ b/src/api.cc
@@ -486,6 +486,10 @@ class ArrayBufferAllocator : public v8::ArrayBuffer::Allocator {
     return data;
   }

+  void* Realloc(void* data, size_t length) override {
+    return realloc(data, length);
+  }
+
   void Free(void* data, size_t) override { free(data); }
 };

 